jitter.py

```diff
@@ -0,0 +1,178 @@
+import epics
+import time
+import argparse
+import csv
+import threading
+from collections import defaultdict
+import sys
+
+
+class JitterMeasurer:
+    def __init__(self, args):
+        self.args = args
+        self.results = []
+        self.measurement_lock = threading.Lock()
+        self.active_measurements = {}
+        self.initial_values = {}
+
+    def generate_pv_names(self):
+        pv_pairs = []
+        for sec in self.args.sector:
+            for subsec in self.args.subsection:
+                for pln in self.args.plane:
+                    cell_str = f"{sec:02d}"
+                    sp = f"{self.args.prefix}SI-{cell_str}{subsec}:PS-{pln}:Kick-SP"
+                    rb = f"{self.args.prefix}SI-{cell_str}{subsec}:PS-{pln}:Current-RB"
+                    pv_pairs.append((sp, rb, sec, subsec, pln))
+        return pv_pairs
+
+    def on_setpoint_change(self, pvname, value, **kwargs):
+        with self.measurement_lock:
+            if pvname not in self.active_measurements:
+                return
+
+            measurement = self.active_measurements[pvname]
+            if not measurement['expecting_sp']:
+                return
+
+            # Only record if value matches the test value
+            if abs(value - measurement['test_value']) < 0.0001:
+                measurement['sp_timestamp'] = time.time()
+                measurement['expecting_sp'] = False
+                measurement['expecting_rb'] = True
+
+    def on_readback_change(self, pvname, value, **kwargs):
+        with self.measurement_lock:
+            if pvname not in self.active_measurements:
+                return
+
+            measurement = self.active_measurements[pvname]
+            if not measurement['expecting_rb']:
+                return
+
+            measurement['rb_timestamp'] = time.time()
+            measurement['expecting_rb'] = False
+            measurement['completed'] = True
+
+    def measure_for_pv(self, sp, rb, sec, subsec, pln):
```

Comment:
Am I correct in assuming this function is ready to be called by multiple different threads, but this isn't done in the current script?